### ISSUE TITLE
Unable to set different device id for different modsim instances.

### ISSUE THREAD

**Issue by evo0705** [sanny32/OpenModSim#18]
Somehow, when changing the device id in ModSim2, it changes the device id in ModSim1 to the same Id.
![image](https://github.com/user-attachments/assets/cd317ca7-620b-4082-8e9d-14ded2538eef)

**Comment by sanny32** [comment]
Hello @evo0705. This is the “feature”. Because Qt Modbus stack doesn’t allow use multiple id's.

**Comment by talenat-st** [comment]
@sanny32 this is really shame as your programm is superb and multiple slaves (addresses) would be really really needed.
I use OpenModSim for testing and find it comparable to ModBus slave except this "feature" that limits use.
Please add this to new release

**Comment by sanny32** [comment]
This will require rewriting the modbus stack from the qt library, which I would not like to do.

**Comment by sanny32** [comment]
Of course, there is no need to rewrite the entire stack. But the functionality responsible for processing incoming modbus requests is in the private Qt class and cannot be redefined from the outside (see https://github.com/qt/qtserialbus/blob/dev/src/serialbus/qmodbustcpserver_p.h#L72). Therefore, there is a need to rewrite the stack in terms of processing all incoming requests. And this entails moving the entire stack of qt inside the program.

**Comment by raspopov** [comment]
> But the functionality responsible for processing incoming modbus requests is in the private Qt class and cannot be redefined from the outside 

This is not entirely true. We can replace ModbusTcpServer's internal object, `d_ptr`, with our own one by using its `swap()` method and re-assembling it with private classes from the Qt source code changing for example the `QModbusTcpServerPrivate::matchingServerAddress()` method.

**Comment by raspopov** [comment]
I would like to express my sincere gratitude for this feature! Debugging complex hardware is much easier now. 😊
